### PR TITLE
Enable overriding the backend url to enable standalone mode

### PR DESCRIPTION
**Pull request by chetanmeh**
This PR enables running api gateway with standalone server (apache/incubator-openwhisk#4571). 

It also fixes current broken build by updating the `opm` version

### Implementation Details

Api Gateway uses the backend url as computed by [makeWebActionBackendUrl][1] in `createApi` call. This backend url is based on url passed by `wsk api` command

```json
{"apidoc":{"namespace":"_","gatewayBasePath":"/hello","gatewayPath":"/world","gatewayMethod":"GET","id":"API:_:/hello","action":{"name":"hello","namespace":"_","backendMethod":"GET","backendUrl":"http://localhost:3233/api/v1/web/_/default/hello.http","authkey":"23bc46b1-71f6-4ed5-8c54-816aa4f8c502:123zO3xZCLrMN6v2BKK1dXYFpXlPkccOFqm12CdAsMgRU4VrNZ9lyGVCGuMDGIwP"}}}
```

With standalone case this url becomes `http://localhost:3233` and is used as is by api gateway routing logic. So to enable this flow to work the `backendRouting` logic now exposes a env variable `BACKEND_HOST`. If this variable is found to be set then it supersedes the backend url as specified in `createApi` call

[1]: https://github.com/chetanmeh/incubator-openwhisk/blob/29d0d5d922d13e662c20b6bec129ade6d30d0140/core/routemgmt/common/apigw-utils.js#L602

### CODE REVIEW

File: scripts/lua/policies/backendRouting.lua [hunk 1]
```diff
@@ -39,6 +40,11 @@ function _M.setRoute(backendUrl, gatewayPath)
     ngx.req.set_uri(getUriPath(u.path))
   end
   ngx.var.backendUrl = backendUrl
```
Comment (rabbah) [inline scripts/lua/policies/backendRouting.lua]: it's ok to not also change ngx.var?

**Comment by chetanmeh** [comment]
@mrutkows @dgrove-oss The build is failing due to scancode issue reporting lots of trailing whitespace in code from downloaded lua modules. Anyway to fix scancode logic to exclude those files?

**Comment by dgrove-oss** [comment]
@chetanmeh -- we can reorder travis to do the scancode before it installs lua.   Do the scancode in preinstall stage.   Do you want to do it as part of this PR, or do you want me to do a separate PR just for that?

File: scripts/lua/policies/backendRouting.lua (hunk 1 above)
Comment (chetanmeh) [reply scripts/lua/policies/backendRouting.lua]: Changed the `ngx.var` also

**Comment by chetanmeh** [comment]
@dgrove-oss Please review the travis and build related changes once

**Comment by mhamann** [comment]
@chetanmeh can you write some tests for this? we need to make sure that this doesn't break production use cases. I'm assuming it does not, but there are no test assertions proving that.

**Comment by rabbah** [comment]
a visual inspection shows all the changes are guarded by `if backendOverride ~= nil then` whose value is derived from a new environment variable, making this low risk in general.

of course, adding tests ensures this new feature isn't broken in the future.

**Comment by mhamann** [comment]
Positive and negative tests are both needed (proving that both cases work as intended).

While I agree that the code looks correct, I've been bitten plenty of times by code that looked fine visually, but had some nasty side-effect nobody noticed.

**Comment by chetanmeh** [comment]
@mhamann Added few tests in #350. Please review

**Pull request merged**